src/util.rs

```diff
@@ -69,32 +67,27 @@ pub fn io_err(msg: &str) -> io::Error {
     io::Error::new(io::ErrorKind::Other, msg)
 }
 
-pub struct OnceCallback<T> {
-    callback: Arc<Mutex<Option<SendBoxFnOnce<(Result<T, ::Error>,)>>>>,
+pub struct StateCallback<T> {
+    callback: Arc<Mutex<Option<Box<dyn Fn(T) + Send>>>>,
 }
 
-impl<T> OnceCallback<T> {
-    pub fn new<F>(cb: F) -> Self
-    where
-        F: FnOnce(Result<T, ::Error>),
-        F: Send + 'static,
-    {
-        let cb = Some(SendBoxFnOnce::from(cb));
+impl<T> StateCallback<T> {
+    pub fn new(cb: Box<dyn Fn(T) + Send + 'static>) -> Self {
```

Comment:
Why does the function in the box need to have a static lifetime?

Reply:
An excellent question, and it looks like it doesn't now. We had to make that live the `'static` life once upon a time because the closure trick `boxfnonce` uses to ensure that the trait works requires that lifetime, since otherwise the compiler couldn't be sure what was inside closure inside the box.

Now that there's actual compiler support for this, that lifetime didn't need to copy up from the where clause, so I can remove it -- yay! Great cathc.